File: app/src/main/java/com/onemoresecret/WiFiPairingFragment.java
package com.onemoresecret;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.onemoresecret.databinding.FragmentWiFiPairingBinding;

public class WiFiPairingFragment extends Fragment {
    private FragmentWiFiPairingBinding binding;

    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        binding = FragmentWiFiPairingBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.txtResponseCode.setVisibility(View.INVISIBLE);
        binding.txtIntroResponse.setVisibility(View.INVISIBLE);
    }

    public void setData(String requestId, String responseCode, Runnable onConfirm) {
        binding.txtRequestId.setText(requestId);
        binding.txtResponseCode.setText(responseCode);
        binding.btnConfirm.setOnClickListener(e -> {
            binding.btnConfirm.setEnabled(false);
            binding.btnConfirm.setText(R.string.pairing_accepted);
            binding.txtResponseCode.setVisibility(View.VISIBLE);
            binding.txtIntroResponse.setVisibility(View.VISIBLE);
            onConfirm.run();
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/onemoresecret/WiFiPairingFragment.kt
package com.onemoresecret

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import com.onemoresecret.composable.OneMoreSecretTheme
import com.onemoresecret.composable.WiFiPairing

class WiFiPairingFragment : Fragment() {
    private var requestIdState by mutableStateOf("")
    private var responseCodeState by mutableStateOf("")
    private var isPairingAcceptedState by mutableStateOf(false)
    private var onConfirmState: () -> Unit = {}

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext()).apply {
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)

            setContent {
                OneMoreSecretTheme {
                    WiFiPairing(
                        requestId = requestIdState,
                        responseCode = responseCodeState,
                        isPairingAccepted = isPairingAcceptedState,
                        onConfirm = {
                            isPairingAcceptedState = true
                            onConfirmState()
                        }
                    )
                }
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        isPairingAcceptedState = false
    }

    fun setData(requestId: String, responseCode: String, onConfirm: Runnable) {
        requestIdState = requestId
        responseCodeState = responseCode
        isPairingAcceptedState = false
        onConfirmState = { onConfirm.run() }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        onConfirmState = {}
    }
}
